### PR TITLE
IGNITE-14292: Change permissions required to create/destroy caches in GridRestProcessor

### PR DESCRIPTION
**Pull request by Sega76**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [x] There is a single JIRA ticket related to the pull request. 
- [x] The web-link to the pull request is attached to the JIRA ticket.
- [x] The JIRA ticket has the _Patch Available_ state.
- [x] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [x] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [x] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/rest-http/src/test/java/org/apache/ignite/internal/processors/rest/protocols/http/jetty/RestProcessorAuthorizationTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+import static org.apache.ignite.internal.processors.cache.CacheGetRemoveSkipStoreTest.TEST_CACHE;
+import static org.apache.ignite.plugin.security.SecurityPermissionSetBuilder.ALLOW_ALL;
+
+/**
+ * Tests REST processor authorization commands GET_OR_CREATE_CACHE / DESTROY_CACHE.
+ */
+public class RestProcessorAuthorizationTest extends CommonSecurityCheckTest {
+    /** */
+    private final List<TestAuthorizationContext> hndlr = new ArrayList<>();
```
Comment (andrey-kuznetsov) [inline modules/rest-http/src/test/java/org/apache/ignite/internal/processors/rest/protocols/http/jetty/RestProcessorAuthorizationTest.java:45]: It's not a handler, please rename. (List of authorizations seen)

File: modules/core/src/test/java/org/apache/ignite/internal/processors/security/impl/TestAuthorizationContextSecurityPluginProvider.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        /** */
+        public TestAuthorizationContext(String name, SecurityPermission perm,
+            SecurityContext securityCtx) {
+            this.name = name;
+            this.perm = perm;
+            this.securityCtx = securityCtx;
+        }
+
+        /** */
+        public String getName() {
```
Comment (andrey-kuznetsov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/security/impl/TestAuthorizationContextSecurityPluginProvider.java:78]: Getters are redundant to me; public final fields are enough.

**Review by andrey-kuznetsov — CHANGES_REQUESTED** [review]

File: modules/rest-http/src/test/java/org/apache/ignite/internal/processors/rest/protocols/http/jetty/RestProcessorAuthorizationTest.java (hunk 1 above)
Comment (Sega76) [reply modules/rest-http/src/test/java/org/apache/ignite/internal/processors/rest/protocols/http/jetty/RestProcessorAuthorizationTest.java]: done

File: modules/core/src/test/java/org/apache/ignite/internal/processors/security/impl/TestAuthorizationContextSecurityPluginProvider.java (hunk 2 above)
Comment (Sega76) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/security/impl/TestAuthorizationContextSecurityPluginProvider.java]: fixed

**Review by andrey-kuznetsov — APPROVED** [review]

**Review by nizhikov — APPROVED** [review]

**Pull request merged**